crates/task-impls/src/consensus.rs

```diff
@@ -1274,12 +1331,27 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 error!("Failed to sign leaf.commit()!");
                 return false;
             };
+
+            let upgrade_cert;
+
+            // Check if the upgrade certificate we have is for the current view.
+            if let Some(cert) = self.upgrade_cert.clone() {
+                if cert.view_number == view {
+                    upgrade_cert = Some(cert);
+                } else {
+                    upgrade_cert = None;
+                }
+            } else {
+                upgrade_cert = None;
+            }
+
```

Comment:
nit can probably remove the clone with something like this:
```suggestion

            let upgrade_cert = self.upgrade_cert.is_some_and(|cert| { cert.view_number == view }) {
                self.upgrade_cert.take()
            } else {
               None
            };

```

Also will leave self.upgrade_cert == None after we use it

Reply:
Changed!